isce2_topsapp/tropo_proc.py

```diff
@@ -0,0 +1,98 @@
+                                 orbit_path=orbit_path
+                                 )
+
+    with open(f'{yml_name}.yml', "w") as file:
+        file.write(raider_yml)
+
+    cmd = f'raiderDelay.py {yml_name}.yml'
+    result = subprocess.run(cmd,
+                            shell=True)
+
+    if result.returncode:
+        raise ValueError('Raider failed; check logs and standard i/o')
+
+    d = acq_datetime.strftime('%Y%m%d')
+    t = acq_datetime.strftime('%H%M%S')
+    out_nc = Path(f'{weather_model_name}_tropo_{d}T{t}_std.nc')
+
+    if not out_nc.exists():
+        raise ValueError(f'Expected netcdf was not generated for given inputs: {out_nc}')
+    return out_nc
+
+
+def compute_tropo_delay_for_insar_pair(*,
+                                       ref_acq_datetime: datetime.datetime,
+                                       sec_acq_datetime: datetime.datetime,
+                                       ref_orbit_path: str | Path,
+                                       sec_orbit_path: str | Path,
+                                       weather_model_name: str,
+                                       bounding_box: list,
+                                       cube_spacing_in_m: float = 2_000,
+                                       output_epsg_number: int = 4326) -> Path:
+
+    ref_delay_cube_path = calculate_tropo_acq_delay(acq_datetime=ref_acq_datetime,
+                                                    orbit_path=ref_orbit_path,
+                                                    weather_model_name=weather_model_name,
+                                                    bounding_box=bounding_box,
+                                                    cube_spacing_in_m=cube_spacing_in_m,
+                                                    output_epsg_number=output_epsg_number,
+                                                    yml_name='reference_raider'
+                                                    )
+
+    sec_delay_cube_path = calculate_tropo_acq_delay(acq_datetime=sec_acq_datetime,
+                                                    orbit_path=sec_orbit_path,
+                                                    weather_model_name=weather_model_name,
+                                                    bounding_box=bounding_box,
+                                                    cube_spacing_in_m=cube_spacing_in_m,
+                                                    output_epsg_number=output_epsg_number,
+                                                    yml_name='secondary_raider'
+                                                    )
+
+    ds_ref = xr.open_dataset(ref_delay_cube_path, engine='rasterio')
+    ds_sec = xr.open_dataset(sec_delay_cube_path, engine='rasterio')
+
+    # Wavelenth of S1 in meters according to
+    # https://sentinels.copernicus.eu/web/sentinel/technical-guides/sentinel-1-sar/sar-instrument
+    lambda_s1 = .055465763
+    ds_delay = (ds_ref - ds_sec) * (lambda_s1) / (4 * np.pi)
```

Comment:
@dbekaert - how does this look?

Note this is doing the differential delay at all levels/variables in the group of the cube (i.e. both `hydro` and `wet`).

Reply:
LGTM @cmarshak .

We could add a bit more complexity. Raider does not do temporal interpolation and i missed to mention that before.
1)  Raider currently rounds to closed available model UTC epic (i.e.  depends on the model) when you provide a satellite UTC time. We could do 4 raider computations to compute the allowed UTC before and after the satellite UTC. 
2) We do the temporal interpolation in time with linear weights for reference  and then for secondary dates.
3) you then compute the interferometric delay like you do above. Although, we should extract wavelength on the fly from within product meta-data so this becomes suitable for putting also in ALOS workflow etc.

Once you have a prototype would be good to do a quick test to the data to see if what is done is correctly. Take a small temporal baseline GUNW (e.g. 6 days) that does _not_ cover a large event like  EQ. 
- Plot A. A = compute unwrapped phase minus mean of the unwrapped phase 
- Plot B on same colorbar as A. B = compute 2D tropospheric correction minus mean of the tropospheric phase. Note to compute the troposheric phase as the sum of the hydrostatic and wet 3D cube intersected with the DEM. The latter should be simple scipy and should exist in ARIA tools and raider already (@sssangha or @jlmaurer)
- plot A-B on same colorbar as A.